ap_src/ap_app/static/js/teams.js

```diff
@@ -53,26 +53,47 @@ function favouriteTeam(e, user, id) {
     })
 }
 
-function LeaveTeam(e, user, redirect) {
-    var data = JSON.stringify({"username": user, "team": e.id})
-    fetch(apiURL + 'api/manage/?method=leave', {
-        method: "post",
-        body: data,
-        headers: {
-            "Content-Type": "application/json",
+/*
+Leaves team but also checks for lingering permissions and removes them
+*/
+async function LeaveTeam(e, user, token) {
+    const headers = {
+      "Content-Type": "application/json",
+      "X-CSRFToken": token,
+    };
+  
+    var data = JSON.stringify({ username: user, team: e.id });
+  
+    const leaveResponse = await fetch(apiURL + "api/manage/?method=leave", {
+      method: "post",
+      body: data,
+      headers: headers,
+    });
+  
+    if (!leaveResponse.ok)
+      throw new Error(`Leave request failed with status ${leaveResponse.status}`);
+  
+    const leaveData = await leaveResponse.json();
+  
+    if (leaveData.message === "success") {
+      const permissionsResponse = await fetch(
+        window.location.origin + "/remove_lingering_perms",
+        {
+          method: "post",
+          body: data,
```

Comment:
It looks like only username is used in remove_lingering_perms. So instead of `body: data` you should be able to just pass `body: username`

Reply:
It turns out there does not need to be any data passed through at all to the second POST request. The data was removed with 3ee4a58

Reply:
See also c23754d